### PR TITLE
Config refactor

### PR DESCRIPTION
**Pull request by henridevieux**
Before opening your pull request, please make sure that you have:

- [x] signed [Uber's Contributor License Agreement][];
- [x] added tests to cover your changes;
- [x] run the test suite locally (`make test`); and finally,
- [x] run the linters locally (`make lint`).

Thanks for your contribution!

[Uber's Contributor License Agreement]: https://docs.google.com/a/uber.com/forms/d/1pAwS_-dA1KhPlfxzYLBqK6rsSWwRwH95OCCZrcsY5rk/viewform

### CODE REVIEW

**Comment by henridevieux** [comment]
This removes the dependency on the environment variable to find the primary yaml config file. Instead, that's now passed via command line and has a default value in defines.go. 

What was previously passed via the command line via the "-c" option was the targets config. That is now defined in the yaml config as standaloneTargetConfig. That file is used when the orchestrator mode is disabled via an added field in the yaml config file as well.

File: internal/log/log.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func CreateLogger(
 		}
 		bootstrapLogger.Info("Log file path provided", zap.String("path", c.LogSink))
 		if c.StdOut || foreground {
-			output = zap.Output(zap.Tee(os.Stdout, sink))
+			output = zap.Output(zap.MultiWriteSyncer(os.Stdout, sink))
```
Comment (henridevieux) [inline internal/log/log.go]: Perhaps jcorbin can provide insight as to whether this is actually the best way to do this.

File: config/config.go [hunk 2]
```diff
@@ -185,16 +169,13 @@ func ParseCliArgs(logger zap.Logger, service string, version string) *CLIConfig
 	app := cli.App(service, "Utility to echo the DC and Cloud Infrastructure")
 
 	app.Version("v version", "Arachne "+version)
-	app.Spec = "[--foreground] [-c=<config_file_path>] [--receiver_only] [--sender_only] [--orchestrator]"
-
-	defTargetLocalPath := path.Join(getConfigDir(), defTargetFile)
+	app.Spec = "[--foreground] [-c=<config_file>] [--receiver_only] [--sender_only]"
```
Comment (Vasilis) [inline config/config.go]: The corresponding section in README needs update.

File: glide.lock [hunk 3]
```diff
@@ -55,7 +55,7 @@ imports:
 - name: github.com/uber-go/atomic
   version: 0c9e689d64f004564b79d9a663634756df322902
 - name: github.com/uber-go/zap
-  version: 8ec65dd53fd33dce854f899316fdb75c7c695fc0
+  version: a5783ee4b216a927da8f839c45cfbf9d694e1467
```
Comment (Vasilis) [inline glide.lock]: This is the result of a glide update?

File: config/config.go (hunk 2 above)
Comment (Vasilis) [inline config/config.go]: Additionally, ParseCliArgs() is not covered in config_test.go

File: glide.lock (hunk 3 above)
Comment (henridevieux) [inline glide.lock]: Oh - I need to fix this, and remove viper and fsnotify from glide

File: defines/defines.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ const (
 	BatchIntervalUploadStats        = 0.95
 	ChannelInBufferSize             = 800
 	ChannelOutBufferSize            = 800
+	DefaultConfigFile               = "/etc/arachne/arachne.yaml"
```
Comment (jcorbin) [inline defines/defines.go]: imo, a better default would be a cwd/pwd-relative "arachne.yaml" or "config.yaml", leaving any absolute paths to be specified outside of code; in other words, default to "only/maybe useful for local development, but all distributors/integrators want to set the flag".

File: glide.lock (hunk 3 above)
Comment (henridevieux) [inline glide.lock]: done

File: config/config.go (hunk 2 above)
Comment (henridevieux) [inline config/config.go]: Done. 
I can add more tests to config_test.go separately.

File: internal/log/log.go [hunk 5]
```diff
@@ -75,7 +75,7 @@ func CreateLogger(
 		}
 		bootstrapLogger.Info("Log file path provided", zap.String("path", c.LogSink))
 		if c.StdOut || foreground {
-			output = zap.Output(zap.MultiWriteSyncer(os.Stdout, sink))
+			output = zap.Output(zap.Tee(os.Stdout, sink))
```
Comment (Vasilis) [inline internal/log/log.go]: Any specific reason for reverting this back  to the older version? I am afraid the `go get` might fail again.

File: defines/defines.go (hunk 4 above)
Comment (Vasilis) [inline defines/defines.go]: Actually, this const is not even needed since the default is already provided by the `default` CLI option.

Comment (henridevieux) [inline defines/defines.go]: Not sure what you mean, @Vasilis  - the CLI defaults to what we define here if --config_file is not passed. Are you suggesting we remove this line and hard code code it in the ParseCliArgs function?

**Pull request merged**